### DISCUSSION TITLE
About Lidgren Network

### DISCUSSION THREAD

**Discussion by FrannDzs** [FakeFishGames/Barotrauma#12191] [Ideas and suggestions]
Use the Lidgren fork of SpaceWizards, which is still under maintenance:
https://github.com/space-wizards/SpaceWizards.Lidgren.Network
instead of 
https://github.com/lidgren/lidgren-network-gen3

Or alternatively ENet which is also still in maintenance but it is a total change
https://github.com/lsalzman/enet

**Comment by Regalis11** [comment]
Thank you for the suggestion! Are you aware of any specific fixes or improvements in the fork that might be relevant in Barotrauma? I can't off the top of my head think of any known issues in Lidgren that affect Barotrauma, and it also seems the performance impact of Lidgren is completely negligible in our case.

**Reply by FrannDzs** [reply]
I have some problems with the dedicated server, I could never publish it in the server list. 

I don't really know if it is useful, although it has the advantage of being maintained, but you can see the changes here
https://github.com/space-wizards/SpaceWizards.Lidgren.Network/blob/wizards/RELEASE-NOTES.md

This releases have a lot of important changes:
- [0.1.0] https://github.com/space-wizards/SpaceWizards.Lidgren.Network/blob/wizards/RELEASE-NOTES.md#010
- [0.2.0] https://github.com/space-wizards/SpaceWizards.Lidgren.Network/blob/wizards/RELEASE-NOTES.md#020

Regards

**Reply by Regalis11** [reply]
> I have some problems with the dedicated server, I could never publish it in the server list.

That's not caused by Lidgren: the server list is implemented using Steamworks, and doesn't use Lidgren in any way.

> I don't really know if it is useful, although it has the advantage of being maintained

Well, in a sense that could be considered a bad thing: Lidgren has been widely used for years and we know it's quite solid. A new fork that's in active development can have new issues introduced by the changes, and if those changes aren't even something that are relevant in our case, changing to that fork could even be detrimental for us.
